compiler/rustc_resolve/src/check_unused.rs

```diff
@@ -494,6 +496,30 @@ impl Resolver<'_, '_> {
         }
 
         let unused_imports = visitor.unused_imports;
+
+        // The lint fixes for unused_import and unnecessary_qualification may conflict.
+        // Deleting both unused imports and unnecessary segments of an item may result
+        // in the item not being found.
+        for v in &self.potentially_unnecessary_qualification {
+            if let LexicalScopeBinding::Item(name_binding) = v.binding
+                && let NameBindingKind::Import { import, .. } = name_binding.kind
+                && let Some(unused_import) = unused_imports.get(&import.root_id)
+                && let Some(id) = import.id()
+                && unused_import.unused.contains(&id)
```

Comment:
Shouldn't redundant imports be excluded here as well?

Reply:
> Shouldn't redundant imports be excluded here as well?

Thank you. yes, I add it now, because there is no fix for redundant import, so I didn’t add it at first.